src/common/WorkQueue.cc

```diff
@@ -122,6 +122,8 @@ void ThreadPool::worker(WorkThread *wt)
 	  processing++;
 	  ldout(cct,12) << "worker wq " << wq->name << " start processing " << item
 			<< " (" << processing << " active)" << dendl;
+          assert(strlen(wq->thread_name) < 16);
+          pthread_setname_np(wt->get_thread_id(), wq->thread_name);
```

Comment:
@liewegas Hmm, earlier you said that it's ok: https://github.com/ceph/ceph/pull/5882#discussion_r45862210 now, you're saying it's not. Following monitor quorum in ceph, you need to answer me once again - this will be the third and deciding vote ;)

I think, adding names for work queues is ok, and changing thread name when wq runs, because when osd/monitor will hang you could faster diagnose it only running top/ps.

I don't get the idea to name the thread after the thread pool, I'm missing something? Can you give me example? Before adding names to wq, there were only two names "thread_pool" and "sharded_th_pool", adding wq names modify threads that was previously called "thread_pool'. So should I revert it and add %d e.g. tp_thpool_1  tp_s_thpool_1 ?